contracts/marketplace/lib.rs

```diff
@@ -181,6 +221,18 @@ pub mod marketplace {
             Ok(())
         }
 
+        /// Update the length of the auction.
+        ///
+        /// Can only be called by an account with an Admin role
+        #[ink(message)]
+        pub fn set_auction_length(&mut self, new_auction_length: BlockNumber) -> Result<(), Error> {
```

Comment:
should it be allowed only for a halted / ended contract? I'm guessing that changing auction length just before some user's transaction might introduce some unexpected and confusing behaviour

Reply:
out of abundance of caution we can add that, sure